### ISSUE TITLE
processTerm for search is called with index as second parameter

### ISSUE THREAD

**Issue by Alxandr** [lucaong/minisearch#5]
```js
mySearch.search('foo bar', {
  processTerm(...args) {
    console.log(args);
  }
});

// logs:
// ['foo', 0, ['foo', 'bar']]
// ['bar', 1, ['foo', 'bar']]
```

**Comment by lucaong** [comment]
You are right, that is because it's used as the argument of `Array.prototype.map`. I will wrap it in a function to prevent people depending on this.

**Comment by Alxandr** [comment]
I was depending on the value being `undefined` (as per the spec I think?). Had to do `typeof arg === 'number' || typeof arg === 'undefined'`.

**Comment by lucaong** [comment]
you are right, this is a bug, as the documentation specifies that it should be undefined. I am about to push a new patch version fixing this (and adding a test).

**Comment by lucaong** [comment]
Fixed in `v1.1.1`
